### PR TITLE
Implement toJson on ReactServerAgent request for logging/debugging

### LINKED ISSUE #228: Implement toJson
```
2016-05-30T04:22:54.628Z - error: [react-server.core.renderMiddleware] Error rendering element Unknown message=toJSON() from superagent's API isn't implemented yet., stack=Error: toJSON() from superagent's API isn't implemented yet.
    at Request.(anonymous function) [as toJSON] (/Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react-server/target/server/ReactServerAgent/Request.js:39:10)
    at Object.stringify (native)
    at new StationList (station-list.js:22:34)
    at [object Object].ReactCompositeComponentMixin.mountComponent (/Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react/lib/ReactCompositeComponent.js:148:18)
    at [object Object].wrapper [as mountComponent] (/Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react/lib/ReactPerf.js:66:21)
    at Object.ReactReconciler.mountComponent (/Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react/lib/ReactReconciler.js:37:35)
    at [object Object].ReactCompositeComponentMixin.mountComponent (/Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react/lib/ReactCompositeComponent.js:225:34)
    at [object Object].wrapper [as mountComponent] (/Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react/lib/ReactPerf.js:66:21)
    at /Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react/lib/ReactServerRendering.js:42:38
    at ReactServerRenderingTransaction.Mixin.perform (/Users/doug.wade/code/react-server/packages/react-server-examples/bike-share/node_modules/react/lib/Transaction.js:136:20)
```

### PR DESCRIPTION
**Pull request by doug-wade**
#228

### CODE REVIEW

File: packages/react-server/core/ReactServerAgent/Request.js [hunk 1]
```diff
@@ -101,6 +101,20 @@ Request.prototype.type = function (type) {
...
+	return {
+		aborted: this.aborted,
+		agent: this._agent,
+		headers: this._headers,
+		method: this._method,
+		postParams: this._postParams,
+		queryParams: this._queryParams,
+		timeout: this._timeout,
+		type: this._type,
+		urlPath: this._urlPath,
```
Comment (doug-wade) [inline packages/react-server/core/ReactServerAgent/Request.js]: I just slapped everything into the json that I could find on `this` -- lemme know if I missed anything.

Comment (gigabo) [inline packages/react-server/core/ReactServerAgent/Request.js]: This LGTM.

@roblg - Do I remember you adding some special handling for this outside of React Server core?

Any conflict here?

Comment (roblg) [inline packages/react-server/core/ReactServerAgent/Request.js]: @gigabo It's possible :grin:. I don't recall anything though. There's some special handling of ReactServerAgent _responses_ in Cache.js (sibling to Request.js), and there's some special handing of error objects in Redfin-internal code, but I don't think I recall any Request-related handling.

@doug-wade there's also a `_cacheWhitelist` property (whose name doesn't make any sense unless you look at the implementation) that includes an array of headers that we're interested in from the response. It's technically state on the request; not sure if it's in interesting in a JSON representation of the request.

Somewhat related, are we sure we want to include the raw `agent` object in a `toJSON` method? If set, it's a node HTTP Agent, which doesn't really seem JSON-y to me (though maybe it has properties we want to expose?)

Comment (gigabo) [inline packages/react-server/core/ReactServerAgent/Request.js]: Nice catch @roblg.  Gotta get your eyes on more PRs here! 👀

Comment (doug-wade) [inline packages/react-server/core/ReactServerAgent/Request.js]: @roblg Thanks!  I've removed `_agent` and added `_cacheWhitelist`

**Pull request merged**